File: app/src/main/java/artwork/java/Preview.java
package artwork.java;

import android.content.Context;
import android.graphics.*;
import android.util.AttributeSet;
import artwork.java.R;

public class Preview extends androidx.appcompat.widget.AppCompatImageView {
    public int w, h;

    /**
     * @param context the context
     */
    public Preview(Context context) {
        super(context);
    }
    public Preview(Context context, AttributeSet attrs) {
        super(context, attrs);
        this.setImageResource(R.drawable.image31);
        //
    }
    public void onSizeChanged(int w, int h, int oldw, int oldh) {
        this.w = w; this.h = h;
    }


    @Override
    public void onDraw(Canvas canvasGlobal) {
        super.onDraw(canvasGlobal);
    }

}

File: app/src/main/java/artwork/kotlin/Preview.kt
package artwork.kotlin

import android.content.Context
import android.graphics.Canvas
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageView
import artwork.java.R

class Preview : AppCompatImageView {
    var w: Int = 0
    var h: Int = 0

    /**
     * @param context the context
     */
    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        this.setImageResource(R.drawable.image31)
        //
    }

    public override fun onSizeChanged(w: Int, h: Int, oldw: Int, oldh: Int) {
        this.w = w
        this.h = h
    }

    public override fun onDraw(canvasGlobal: Canvas) {
        super.onDraw(canvasGlobal)
    }
}
